### DISCUSSION TITLE
What would be required to support AsciiDoc(tor) as an alternative input format?

### DISCUSSION THREAD

**Discussion by bentolor** [estruyf/doctor#140] [Ideas]
Hello @estruyf and thanks for taking a lead in unlocking Documentation-as-Code approach viable for MS Sharepoint-based work environments.

I'm really regularly revisiting the topic _"How to crosspost DaC content into a management-acceptable CMS"_ on a bi-yearly basis. And while in my last organization they used Confluence and I was eventually able to use Asciidoctor with Confluence in a working space, now I'm in a Sharepoint-based environment and it seems to me that he Sharepoint-world is lagging at least 5 years behind. So great so see that with Doctor a tools has become available. Thanks for that!

Regarding source format: I'm a big fan and promoter of Asciidoc, as it offers the expressiveness you need for really common tasks like *Tables, Document Metadata, Chapterlinks, Automatic TOCs, File Includes, Callouts* and many more.

I understand that Doctor has taken the road of the most prominent Markup format. But what I miss is what would it take to extend it to i.e. support other formats like Asciidoc(tor) as well: Is the input processing tightly coupled or was the input source designed in a pluggable manner (as the homepage suggest to me on reading _Doctor allows developers to use tools/applications they are used to, like VSCode and Markdown_?

I'm just trying to understand my options here.

Not sure if you know about Asciidoctor: There are Ruby, Java and Javascript implementations available. The most challenging aspect in our Confluence implementation was rather the link and multi-file management, i.e. linking to images, between sections/chapters and the general approach of multiple input and output files vs. the single-page approach of Markdown.

Again, thanks for providing this tool to the community!

**Comment by estruyf** [comment]
Doctor makes use of the OOTB Markdown web part in SharePoint, the problem with any other format is that it needs to be supported in SharePoint. To support Asciidoc, it requires to have a webpart that supports the format. I don't believe there is any webpart that currently supports it.

You can, of course, always create your own webpart and make use of that.

**Comment by bentolor** [comment]
Got it. Thanks for the explanation, @estruyf ! 
This issue can be closed at your convenience.
